src/Type/WebMozartAssert/AssertTypeSpecifyingExtension.php

```diff
@@ -156,50 +158,69 @@ static function (Type $type) {
 			);
 		}
 
-		$expression = self::createExpression($scope, $staticMethodReflection->getName(), $node->getArgs());
-		if ($expression === null) {
+		[$expr, $rootExpr] = self::createExpression($scope, $staticMethodReflection->getName(), $node->getArgs());
+		if ($expr === null) {
 			return new SpecifiedTypes([], []);
 		}
 
-		return $this->typeSpecifier->specifyTypesInCondition(
+		$specifiedTypes = $this->typeSpecifier->specifyTypesInCondition(
 			$scope,
-			$expression,
-			TypeSpecifierContext::createTruthy()
+			$expr,
+			TypeSpecifierContext::createTruthy(),
+			$rootExpr
 		);
+
+		if ($rootExpr !== null) {
+			// Makes consecutive calls with a rootExpr adding unknown info via FAUX_FUNCTION evaluate to true
+			$specifiedTypes = $specifiedTypes->unionWith(
+				$this->typeSpecifier->create($rootExpr, new ConstantBooleanType(true), TypeSpecifierContext::createTruthy())
```

Comment:
I wonder of this can/should be done in phpstan itself. Basically we could specify the root expression with true/false according to context I think.

Reply:
Did not work as expected, see https://github.com/phpstan/phpstan-src/pull/1516